ENH: Automate updates of CHANGES

oesteban opened the pull request:
Ref. https://github.com/nipy/nipype/issues/2313#issuecomment-365330143, item 5.

Example of resulting CHANGES at the moment, running `bash tools/update_changes.sh 1.0.1`:

```diff
diff --git a/CHANGES b/CHANGES
index 8547a5f..1822047 100644
--- a/CHANGES
+++ b/CHANGES
@@ -1,6 +1,32 @@
 Upcoming release
 ================

+1.0.1 (February 13, 2018)
+=========================
+
+  * ENH+FIX: Add 3dTproject AFNI interface, Fix OneDToolPy, Add -noFDR flag to 3dDeconvolve (#2426)
+  * FIX: Update pybids data directory, unbundle out-of-date numpydoc (#2437)
+  * fix: set _id property of cloned node too (#2432)
+  * enh: c3d + c4d interface (#2430)
+  * enh: improve travis build reliability (#2433)
+  * [FIX] Fix issue #2408 (#2429)
+  * fix: updated custom search to current api [skip ci] (#2425)
+  * ENH Allow input weight for AFNI's volreg. (#2396)
+  * ENH: Re-enable spm.Realign to take lists of lists of files (#2409)
+  * [ENH] Delay crashing if exception is raised in local hash check (#2410)
+  * fix little out_file  bugs in  afni.Zcat and afni.Merge (#2424)
+  * fix: remove deprecated output from _list_outputs (#2422)
+  * FIX: argstr for mask in BlurToFWHM (#2418)
+  * FIX: Default value for sbatch_args (SLURMGraph) (#2417)
+  * FIX: PipeFunc __repr__ format string (#2407)
+  * FIX: fix ortvec argstr for Deconvolve (#2415)
+  * [ENH] Add tests to ``hash_files`` (#2405)
+  * [FIX] Make ignore_exception a class attribute (#2414)
+  * Revert "[FIX,WIP] Setting inputs.ignore_exception" (#2413)
+  * [FIX,WIP] Setting inputs.ignore_exception (#2412)
+  * [FIX] Bug fixes for afni.model (#2398)
+  * maint: move out of release mode (#2403)
+
 1.0.0 (January 24, 2018)
 ========================
```

Of course the resulting CHANGES will need some curation, but we won't miss PRs out. Let me know if you want me to also add the URL before the PR handles.

oesteban commented:
BTW, using this the "Upcoming release" header does not make sense anymore.

effigies commented:
One model to consider for that is the numpy approach: https://github.com/numpy/numpy/tree/master/doc/changelog

satra commented:
@effigies - unless it's super easy to implement, perhaps we can do this with next month's release.

effigies commented:
That's fine with me. I'm okay with the current change, but I agree that we can go ahead and drop the "Next release" section.

satra commented:
@oesteban - adding the url would be good

effigies commented:
@oesteban, do you have a couple minutes to make these two changes? I can probably do it if you're swamped.

oesteban commented:
As for the organization of the changelog, we could just rotate everything before 1.0.0 to a new changelog file, link it from the documentation and keep working on a new one. When we jump to 2.0, we can rotate again. I think rotating with minor and patches may be too much.

effigies commented:
Fine with me. We can open a new issue/PR to do any arguing, anyway. Could you paste in a new diff, just for the sake of review?

oesteban commented:
```Diff
diff --git a/CHANGES b/CHANGES
index 56a5fc749..c757f5b97 100644
--- a/CHANGES
+++ b/CHANGES
@@ -1,3 +1,37 @@
+1.0.1 (February 20, 2018)
+=========================
+
+  * [ENH/WIP] Add SPM Fieldmap Tool wrapper (https://github.com/nipy/nipype/pull/1905)
+  * Fix/tests (https://github.com/nipy/nipype/pull/2451)
+  * CI: Add doctests to pytest script (https://github.com/nipy/nipype/pull/2449)
+  * [WIP] COV: Ignore tests in calculating coverage (https://github.com/nipy/nipype/pull/2443)
+  * bug for datagrabber (https://github.com/nipy/nipype/pull/1915)
+  * RF: Update and factor mount table parsing (https://github.com/nipy/nipype/pull/2444)
+  * TST/CI: Split Circle workflow (https://github.com/nipy/nipype/pull/2386)
+  * ENH+FIX: Add 3dTproject AFNI interface, Fix OneDToolPy, Add -noFDR flag to 3dDeconvolve (https://github.com/nipy/nipype/pull/2426)
+  * FIX: Update pybids data directory, unbundle out-of-date numpydoc (https://github.com/nipy/nipype/pull/2437)
+  * fix: set _id property of cloned node too (https://github.com/nipy/nipype/pull/2432)
+  * enh: c3d + c4d interface (https://github.com/nipy/nipype/pull/2430)
+  * enh: improve travis build reliability (https://github.com/nipy/nipype/pull/2433)
+  * [FIX] Fix issue #2408 (https://github.com/nipy/nipype/pull/2429)
+  * fix: updated custom search to current api [skip ci] (https://github.com/nipy/nipype/pull/2425)
+  * ENH Allow input weight for AFNI's volreg. (https://github.com/nipy/nipype/pull/2396)
+  * ENH: Re-enable spm.Realign to take lists of lists of files (https://github.com/nipy/nipype/pull/2409)
+  * [ENH] Delay crashing if exception is raised in local hash check (https://github.com/nipy/nipype/pull/2410)
+  * fix little out_file  bugs in  afni.Zcat and afni.Merge (https://github.com/nipy/nipype/pull/2424)
+  * fix: remove deprecated output from _list_outputs (https://github.com/nipy/nipype/pull/2422)
+  * FIX: argstr for mask in BlurToFWHM (https://github.com/nipy/nipype/pull/2418)
+  * FIX: Default value for sbatch_args (SLURMGraph) (https://github.com/nipy/nipype/pull/2417)
+  * FIX: PipeFunc __repr__ format string (https://github.com/nipy/nipype/pull/2407)
+  * FIX: fix ortvec argstr for Deconvolve (https://github.com/nipy/nipype/pull/2415)
+  * [ENH] Add tests to ``hash_files`` (https://github.com/nipy/nipype/pull/2405)
+  * [FIX] Make ignore_exception a class attribute (https://github.com/nipy/nipype/pull/2414)
+  * Revert "[FIX,WIP] Setting inputs.ignore_exception" (https://github.com/nipy/nipype/pull/2413)
+  * [FIX,WIP] Setting inputs.ignore_exception (https://github.com/nipy/nipype/pull/2412)
+  * [FIX] Bug fixes for afni.model (https://github.com/nipy/nipype/pull/2398)
+  * maint: move out of release mode (https://github.com/nipy/nipype/pull/2403)
+
+
 1.0.0 (January 24, 2018)
 ========================
```

effigies commented on tools/update_changes.sh:
```diff
@@ -0,0 +1,42 @@
+    tail -n+4 CHANGES >> newchanges
+else
+    cat CHANGES >> newchanges
+fi
```
Maybe let's just remove this `if`/`else` block. It doesn't seem valuable to re-add "Upcoming release", even if it's present.

oesteban replied:
I've modified the if/else block that was actually re-adding the "Upcoming release" bit.

effigies commented:
LGTM.
